### PR TITLE
OHRM-1049:Form Validation Added

### PR DESCRIPTION
**Pull request by Super-Chama**
OXD Form validation logic added
OXD Button type prop renamed, emits removed to support form submission

**[nanoid](https://www.npmjs.com/package/nanoid) package added**

### CODE REVIEW

File: components/src/core/components/InputField/__tests__/input-field.spec.ts [hunk 1]
```diff
@@ -1,38 +1,69 @@
 import {mount} from '@vue/test-utils';
 import InputField from '@orangehrm/oxd/core/components/InputField/InputField.vue';
+import Form from '@orangehrm/oxd/core/components/Form/Form.vue';
 
 describe('InputField.vue', () => {
+  const mockForm = Form;
+  mockForm.$el = {
+    // eslint-disable-next-line
+    addEventListener(event: any, callback: any) {},
```
Comment (RajithaKumara) [inline components/src/core/components/InputField/__tests__/input-field.spec.ts:9]: Can't we improve
```
addEventListener(event: Event, callback: () => void)
# OR
addEventListener(event: Event, callback: (arg:string) => <return-type>)
```

File: components/package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
   "dependencies": {
     "bootstrap-icons": "^1.3.0",
     "core-js": "^3.6.5",
+    "nanoid": "^3.1.21",
```
Comment (RajithaKumara) [inline components/package.json:13]: @samanthajayasinghe can you check this?

File: components/src/core/components/Button/__tests__/button.spec.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ describe('Button.vue', () => {
   it('renders OXD button', () => {
     const label = 'Button';
     const wrapper = mount(Button, {
-      props: {label},
+      props: {label, type: 'button'},
```
Comment (RajithaKumara) [inline components/src/core/components/Button/__tests__/button.spec.ts:31]: Can't we define the default type of button as `button`

File: components/src/mixins/uuid/index.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+export const uuid = defineComponent({
+    data() {
+        return {
+            cid: "",
+        };
+    },
+    created() {
+        this.cid = nanoid();
+    }
+});
```
Comment (RajithaKumara) [inline components/src/mixins/uuid/index.ts:13]: Add new line end of the file

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: components/src/core/components/Button/__tests__/button.spec.ts (hunk 3 above)
Comment (Super-Chama) [reply components/src/core/components/Button/__tests__/button.spec.ts]: added in base component. can be overridden if needed.

File: components/src/mixins/uuid/index.ts (hunk 4 above)
Comment (Super-Chama) [reply components/src/mixins/uuid/index.ts]: added

File: components/package.json (hunk 2 above)
Comment (samanthajayasinghe) [reply components/package.json:13]: seems MIT license

File: storybook/stories/core/components/Form/ValidatableFrom.story.vue [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  <oxd-divider />
+
+  <oxd-form @submit="checkForm" @submitValid="getFormValues" ref="form">
+    <oxd-form-row>
+      <oxd-form-group class="orangehrm-bottom-space">
+        <oxd-input-field
+            label="Job Category Name"
+            v-model="name"
+            :rules="rules.name"
```
Comment (samanthajayasinghe) [inline storybook/stories/core/components/Form/ValidatableFrom.story.vue:12]: How we add multiple rules . min/max chars ?

Comment (Super-Chama) [reply storybook/stories/core/components/Form/ValidatableFrom.story.vue:12]: Sure, Here we reference the `rules` object. under the key `name` there are mutliple rules, required and max chars.

```
rules {
        name: [
          v => (!!v && v.trim() !== "") || "Required",
          v => (v && v.length <= 50) || "Should be less than 50 characters"
        ],
       ...
}
```
And if we want min chars added we may define it as well.
`v => (v && v.length >= 10) || "Should more than 10 characters"`

**Pull request merged**